connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java

```diff
@@ -147,10 +168,6 @@ public void setUp() {
 
     @Test
     public void testStartStop() throws Exception {
-        expectStart();
```

Comment:
Don't we lose coverage here? The existing test verifies that the log invokes its `createProducer` and `createConsumer` methods exactly once on startup. We should keep those guarantees (especially since those methods have become pluggable over time).

One way to do this is to add tracking logic to the `MockedKafkaBasedLog` class for how many times its `createProducer` and `createConsumer` methods are invoked, and then verify that the number of invocations matches an expected value in the `expectStart` method.